creme/reports/tests/test_views.py

```diff
@@ -0,0 +1,192 @@
+from json import loads as json_load
+from unittest import mock
+
+from django.urls import reverse
+from parameterized import parameterized
+
+from creme.creme_core.auth.entity_credentials import EntityCredentials
+from creme.creme_core.models import SetCredentials
+from creme.reports.report_chart_registry import (
+    ReportPieChart,
+    report_chart_registry,
+)
+
+from .base import BaseReportsTestCase, skipIfCustomReport
+
+
+@skipIfCustomReport
+class GraphFetchSettingsTestCase(BaseReportsTestCase):
+    @classmethod
+    def setUpClass(cls):
+        super(GraphFetchSettingsTestCase, cls).setUpClass()
+
+        report_chart_registry.register(
+            ReportPieChart(name='fakepie', label='Fake Pie')
+        )
+
+    def test_update_settings__missing_id(self):
+        self.login()
+        self.assertPOST404(
+            path=reverse('reports__update_graph_fetch_settings', args=(99999,)),
+            data={
+                "chart": "fakepie",
+            }
+        )
+
+    @mock.patch('creme.reports.views.graph.logger')
+    def test_update_settings__not_allowed(self, mock_logger):
+        """Edition on reports is needed to update the settings"""
+        self.login(is_superuser=False, allowed_apps=['reports'])
+        SetCredentials.objects.create(
+            role=self.role,
+            value=EntityCredentials.VIEW,  # EntityCredentials.CHANGE
+            set_type=SetCredentials.ESET_OWN,
+        )
+
+        graph = self._create_documents_rgraph(user=self.other_user)
+        self.assertEqual(graph.asc, True)
+        self.assertEqual(graph.chart, None)
+
+        response = self.assertPOST200(
+            path=reverse('reports__update_graph_fetch_settings', args=(graph.pk,)),
+            data={
+                "sort": "DESC",
+                "chart": 'fakepie',
+            }
+        )
+
+        self.assertEqual({
+            "sort": "ASC",
+            "chart": None
+        }, json_load(response.content))
```

Comment:
assertJSONEqual() ? response.json() ?
(I do not comment other parts of this file which deserve the same comment)